test/TestProjects/FirstTest-TypeSpec/src/Generated/tspCodeModel.json

```diff
@@ -748,19 +748,146 @@
   },
   {
    "$id": "102",
+   "Name": "Element",
+   "Namespace": "FirstTestTypeSpec",
+   "Description": "element",
+   "IsNullable": false,
+   "Usage": "None",
+   "Properties": [
+    {
+     "$id": "103",
+     "Name": "extension",
+     "SerializedName": "extension",
+     "Description": "",
+     "Type": {
+      "$id": "104",
+      "Name": "Array",
+      "ElementType": {
+       "$id": "105",
+       "Name": "Extension",
+       "Namespace": "FirstTestTypeSpec",
+       "Description": "extension",
+       "IsNullable": false,
+       "Usage": "Input",
+       "BaseModel": {
+        "$ref": "102"
+       },
+       "Properties": [
+        {
+         "$id": "106",
+         "Name": "level",
+         "SerializedName": "level",
+         "Description": "",
+         "Type": {
+          "$id": "107",
+          "Name": "int8",
+          "Kind": "Int32",
+          "IsNullable": false
+         },
+         "IsRequired": true,
+         "IsReadOnly": false
+        }
+       ]
+      },
+      "IsNullable": false
+     },
+     "IsRequired": false,
+     "IsReadOnly": false
+    }
+   ]
+  },
+  {
+   "$ref": "105"
+  },
```

Comment:
this is exactly the result we would like to see in the emitter, and in our yesterday's discussion, it does not require a change in the converter to work properly.